From: Thomas Renninger <trenn@suse.de>
Subject: ACPI: Cleanup acpi_initrd_override declaration and remove ifdefs
References: none
Patch-Mainline: Submitted

Move the acpi_initrd_override() declaration out of CONFIG_ACPI area so that
acpi_initrd_override is also defined if CONFIG_ACPI is not defined.

Move the acpi_initrd_override() call into reserve_initrd() inside the
CONFIG_BLK_DEV_INITRD block.

Now ifdefs around the call in setup.c can be removed.

Cleanup only, no functional change.

Signed-off-by: Thomas Renninger <trenn@suse.de>
Automatically created from "patches.arch/010-acpi_initrd_override_tables.patch" by xen-port-patches.py

--- 13.1.orig/arch/x86/kernel/setup-xen.c	2013-10-21 12:07:06.000000000 +0200
+++ 13.1/arch/x86/kernel/setup-xen.c	2013-10-21 12:08:47.000000000 +0200
@@ -474,6 +474,8 @@ static void __init reserve_initrd(void)
 	relocate_initrd();
 
 	memblock_free(ramdisk_image, ramdisk_end - ramdisk_image);
+
+	acpi_initrd_override((void *)initrd_start, initrd_end - initrd_start);
 }
 #else
 static void __init early_reserve_initrd(void)
@@ -1260,10 +1262,6 @@ void __init setup_arch(char **cmdline_p)
 
 	reserve_initrd();
 
-#if defined(CONFIG_ACPI) && defined(CONFIG_BLK_DEV_INITRD)
-	acpi_initrd_override((void *)initrd_start, initrd_end - initrd_start);
-#endif
-
 #ifndef CONFIG_XEN
 	reserve_crashkernel();
 
